mautic/mautic
# Add support for sms replies as a campaign decision

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | y
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This adds support to consume SMS replies through campaigns

#### Steps to test this PR:
You have to register a webhook with the service provider. For Twilio, it's done under Messaging for the phone number. The callback is `https://your.mautic.net/sms/twilio/callback`

1. Configure a SMS gateway 
2. Create a campaign with a "send text message" action then a "sends a text message" decision and do something as a result
3. Reply to a SMS 
4. The action attached to the decision should have been executed
5. Repeat to test the pattern support (use one that matches, doesn't match, etc)

#### List deprecations along with the new alternative:
1. \Mautic\SmsBundle\Api\AbstractSmsApi class is deprecated. Use \Mautic\SmsBundle\Sms\TransportInterface instead.
2. \Mautic\SmsBundle\Api\TwilioApi is deprecated. Use \Mautic\SmsBundle\Integration\Twilio\TwilioTransport instead
